python/pyarrow/__init__.py

```diff
@@ -24,7 +24,7 @@
    # package is not installed
     try:
         import setuptools_scm
-        __version__ = setuptools_scm.get_version('../')
+        __version__ = setuptools_scm.get_version(root='../../', relative_to=__file__)
```

Comment:
@xhochy and @wesm , I needed to change this because it would only give a version if run under ARROW_HOME/python directory.  So when running Spark tests, on importing pyarrow it would return `None`.  Making it relative to the `__file__` seemed to fix it for all cases.  I can make this a separate JIRA if you think that would be better.

Reply:
See the suggestion above to get rid of this line.